pool.go

```diff
@@ -239,46 +253,60 @@ func (p *Pool) Start(closeFunc func()) {
 			}
 			log.Info("finish closing pool")
 
+			// wait for all on-fly retryable jobs are inserted to db
+			p.retryableWaitGroup.Wait()
```

Comment:
Just theorectically, I think here the synchronization seems incorrect. We expect that the Wait() is called after all goroutines that run `PrepareRetryableJob` calls `Add` but there is no guaranteed about that.
What about calling `p.retryableWaitGroup.Add(1)` in `case job := <-p.RetryJobChan:` before starting the goroutine?